File: Data Structures & Algorithms/maximum-product-subarray/submission-1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res=max(nums)
        curMin,curMax = 1,1
        for n in nums:
            if n == 0:
                curMin = 1
                curMax = 1
                continue
            else:
                tmp = n*curMax
                curMax = max(curMax*n,curMin*n,n)
                curMin = min(tmp,curMin*n,n)
                res = max(res,curMax)
        return res

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int min =nums[0];
        int low = 0,high =nums.length-1;
        while(low<high)
        {
            if(nums[low]>nums[low+1] )
                return nums[low+1];
            else if ( nums[high]<nums[high-1])
                return nums[high];
            else
            {
                low++;
                    high--;
            }

        }
        return min;
    }
}
